pkg/util/imageparser/lifted_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imageparser
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTagRegexp(t *testing.T) {
+	validTags := []string{
+		"latest",
+		"v1.0.0",
+		"v1.0.0-alpha",
+		"my-tag",
+	}
+
+	for _, tag := range validTags {
+		assert.True(t, TagRegexp.MatchString(tag), "expected tag '%s' to be valid", tag)
+	}
+}
+
+func TestDigestRegexp(t *testing.T) {
+	validDigests := []string{
+		"sha256:0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+		"sha384:0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+	}
+
+	for _, digest := range validDigests {
+		assert.True(t, DigestRegexp.MatchString(digest), "expected digest '%s' to be valid", digest)
+	}
+}
```

Comment:
I think maybe we don't need those test cases to maintain these two expressions, and they can be covered by the test in `TestAnchored` function.